daml/Chat.daml

```diff
@@ -17,14 +17,32 @@ template Operator
       nonconsuming OperatorInviteUser : ContractId UserInvitation
         with
           user : Party
+          userName : Text
```

Comment:
I think the Python can be simplified quite a bit more, and more of this move to DAML:

```python
@client.ledger_created('Chat.UserSession')
def handle_session(event):
    return exercise(event.cid, 'UserSessionAccept')
```

```hs
template UserSession
  with
    operator : Party
    user : Party
    userName : Text
  where
    signatory user
    key user : Party
    maintainer key

    controller operator can
      UserSessionAccept : ContractId User
        userCid <- lookupByKey
        case userCid where
          Some(u) -> return u
          None ->
            let contacts = M.fromList [(operator, "Operator")]
          create AddressBook with ..
          create Chat
            with creator = operator, members = [operator, user],
                    name = "Operator Bot",
                    topic = Some "Hey there! I am the Operator Bot",
                    chatId = partyToText user, isPublic = False, ..
          create User with ..
          
      UserSessionArchive : ()
        do return ()
```